src/operations/blas3/gemm_load_store.hpp

```diff
@@ -128,24 +127,20 @@ struct Packetize {
    * @tparam internal True if the current block is internal and no bounds
    * checking is required.
    * @tparam ld The leading dimension of the destination memory. */
-  template <bool trans, bool internal, int ld, typename packet_t = PacketType,
-            typename SrcPointerType, typename DestPointerType, typename index_t,
-            typename EdgePredicate, bool align = aligned>
+  template <bool trans, bool internal, int ld, typename SrcPointerType,
+            typename DestPointerType, typename EdgePredicate,
+            bool align = aligned>
   static SYCL_BLAS_INLINE
       typename std::enable_if<(internal && !trans && align)>::type
-      load(const bool in_range, SrcPointerType src, const index_t &src_offset,
-           DestPointerType dest, const index_t &dest_offset,
+      load(const bool in_range, SrcPointerType src, DestPointerType dest,
            EdgePredicate edge_in_range) {
     if (in_range) {
-      *reinterpret_cast<packet_t *>(
-          static_cast<value_t *>(dest + dest_offset)) =
-          *reinterpret_cast<packet_t *>(
-              static_cast<value_t *>(src + src_offset));
+      *reinterpret_cast<PacketType *>(static_cast<value_t *>(dest)) =
+          *reinterpret_cast<PacketType *>(static_cast<value_t *>(src));
```

Comment:
Why are there both the reinterpret and static cast, wouldn't a single `reinterpret_cast` suffice?

Reply:
I guess that's from older code, but I see it in a few other places as well

Reply:
In this context dest and src are multi_ptrs to value_t, which can't be directly cast to PacketType * without first being cast to a regular value_t *. These functions are removed by #215 anyway!